pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -34,48 +33,34 @@ var _ asset.WritableAsset = (*AgentPXEFiles)(nil)
 // Dependencies returns the assets on which the AgentPXEFiles asset depends.
 func (a *AgentPXEFiles) Dependencies() []asset.Asset {
 	return []asset.Asset{
-		&Ignition{},
-		&BaseIso{},
+		&BaseImage{},
 	}
 }
 
 // Generate generates the image files for PXE asset.
 func (a *AgentPXEFiles) Generate(dependencies asset.Parents) error {
-	ignition := &Ignition{}
-	dependencies.Get(ignition)
 
-	baseImage := &BaseIso{}
+	baseImage := &BaseImage{}
 	dependencies.Get(baseImage)
 
-	a.isoPath = baseImage.File.Filename
+	a.isoPath = baseImage.baseISOFileName
```

Comment:
This, and in particular `extractPXEFileFromISO`, doesn't seem necessary anymore: the dependent asset `BaseImage` already extracted all the base iso content in a temporary folder (see `BaseImage.tmpPath`), so its content could just be simply fetched from there

Reply:
Right, I see the rootfs and kernel also gets extracted to a tmp dir